### ISSUE TITLE
Infinite loop when trying to rebuild a campaign

### ISSUE THREAD

**Issue by adonaicruz** [mautic/mautic#6393]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? |  Yes
| Feature request? | No
| Enhancement? |  No

## Description:
After update to v2.14.0,  some campaigns entered into an **infinite loop** when running the cronjob **mautic:campaigns:rebuild**.

I have just 500 contacts in this segment, for a total of 3000 contacts

Here is an example log:
```
Rebuild leads for list 6
0 total lead(s) to be added in batches of 300
0 total lead(s) to be removed in batches of 300

Rebuild leads for list 7
1 total lead(s) to be added in batches of 300
31126/31126 [============================] 100%
```
![Campaign Print](http://amazon.tekoa.floripa.br/download/print-01.jpg "Campaign Print")

## If a bug:

| Q   | A
| --- | ---
| Mautic version | v2.14.0
| PHP version | 7.1.17

### Steps to reproduce:
1. I could not find a way to reproduce the error in another campaign
 
### Log errors: 
no related errors were generated in [mautic root]/app/log/ or [apache|php]/log/error_log

**Comment by lucasexe** [comment]
I have had the same problem. My server's CPU usage increased to 100%.

**Comment by tekoaweb** [comment]
@lucasexe  use 
`php /var/www/mkt.netcombomais.com.br/app/console mautic:campaigns:rebuild --batch-limit=300 --**max-contacts=5000** >> /dev/null 2>&1` to prevent a infinite loop

I checked witch this query the contacts that are locked:
``
SET @cam=6,@seg=1;
SELECT 
	distinct(ll.lead_id) as id
FROM 
	lead_lists_leads as ll
WHERE
	ll.manually_removed = 0
	AND ll.leadlist_id = @seg
	AND NOT EXISTS (SELECT null FROM campaign_leads as cl WHERE cl.campaign_id = @cam AND cl.lead_id = ll.lead_id);
``

**Comment by lucasexe** [comment]
@tekoaweb 
thank you so much for the solution. I'll try it.

**Comment by tekoaweb** [comment]
@lucasexe it's just a way to not crash your server. I keep trying to solve the problem.

**Comment by michaelbdana** [comment]
same problem here. I have one campaign that causes an infinite loop.

**Comment by adonaicruz** [comment]
giving continuity to what I said in my other account @tekoaweb
while we can not fix the bug, here's how to manually add the loked lead and rebuild the campaign.
that worked for me.
SET @cam=18,@seg=22;
SET @lead=16995;
INSERT INTO `campaign_leads` (`campaign_id`, `lead_id`, `date_added`,`rotation`) VALUES (@cam, @lead,NOW(),1);

**Comment by fuschini** [comment]
I had the same problem. Cloning the campaign and using the new one fixed the issue for me.

**Comment by alanhartless** [comment]
This should prevent the loop: https://github.com/mautic/mautic/pull/6463

**Comment by npracht** [comment]
The related PR has been merged in 2.14.1. Closing the issue now.
